### DISCUSSION TITLE
Bug: 2FA TOTP is not woking

### DISCUSSION THREAD

**Discussion by usamashafiqs** [dani-garcia/vaultwarden#3895] [Q&A]
### Subject of the issue
2FA TOTP is not woking

### Deployment environment

* vaultwarden version: 1.29.2
* Install method: Docker
* Reverse proxy and version: Traefik v2.10.4
### Steps to reproduce
All of TOTP verification code are not working.

**Comment by stefan0xC** [comment]
Can you post the generated support string from the diagnostics page in the [`/admin` panel](https://github.com/dani-garcia/vaultwarden/wiki/Enabling-admin-page)? Most likely your server time is off.

**Reply by BlackDex** [reply]
And probably, when that is posted we will see that the time of the server (or clients) is probably out off sync.
You need to fix that before Vaultwarden can actually work correctly and validate the MFA/2FA tokens.

It also can cause issues with the login tokens and invite tokens, since they are time/date bound, and if those expire or even in the future for the server that could cause issues too.

**Comment by fab-kaz** [comment]
hello,
any news ?
regards
f.
